setup.py

```diff
@@ -23,6 +23,39 @@
           " install setuptools).", file=sys.stderr)
     sys.exit(1)
 
+
+def validate_install_ansible_base():
+    """Validate that we can install ansible-base. Currently this only
+    cares about upgrading to ansible-base from ansible<2.10
+    """
+    # Make sure `lib` isn't in `sys.path` that could confuse this
+    # Later we actually add `lib` to `sys.path`, this is just a precaution
+    sys_modules = set(sys.modules)
+    sys_path = sys.path[:]
+    abspath = os.path.abspath
+    sys.path[:] = [p for p in sys.path if abspath(p) != abspath('lib')]
+
+    try:
+        from ansible.release import __version__
+    except ImportError:
+        pass
+    else:
+        version_tuple = tuple(int(v) for v in __version__.split('.')[:2])
+        if version_tuple < (2, 10):
+            stars = '*' * 76
+            raise RuntimeError(
+                '\n\n%s\n\nCannot upgrade ansible==%s to ansible-base. You must first uninstall\n'
```

Comment:
Maybe add "ansible versions below 2.10" or something so it's clear this is a one-time problem, not that they can never upgrade `ansible` again...

Reply:
I added an extra sentence explicitly detailing this.  Still open to any other things.